src/c++/perf_analyzer/genai-perf/genai_perf/llm_inputs/llm_inputs.py

```diff
@@ -311,6 +318,26 @@ def _add_rows_to_generic_json(
 
         return generic_input_json
 
+    @classmethod
+    def _get_input_dataset_from_file(cls, input_filename: Path) -> Dict:
+        cls.verify_file(input_filename)
+        input_file_prompt = cls._get_prompt_from_input_file(input_filename)
+        dataset_json: Dict[str, Any] = {}
+        dataset_json["features"] = [{"name": "text_input"}]
+        dataset_json["rows"] = []
+        dataset_json["rows"].append({"row": {"text_input": input_file_prompt}})
+        return dataset_json
```

Comment:
Like can the user specify other inputs with this feature? Max-tokens, min-tokens, exclude_input_in_output, etc... it looks like this feature has an unspecified limitation that it only supports text_input. I may be misunderstanding what the user is passing and what's supported here though.

If the user passes a file like the generated llm_inputs.json, which could have other fields (e.g. max_tokens), what happens today? Do those fields get silently ignored?

Reply:
I think i understand.
The flow can be kinda annoying, which is why we wrote the tool initially so users didnt have to deal with this.
Basically there are 3 different inputs sources today: synthetic, dataset, and now file. There are 4 output endpoints today: triton + trtllm, triton + vllm, openai chat, openai completions. 
That is 12 possible paths right now with more to come. So the way we handle this is input_source -> generic json -> specific json. 

The lines you are highlighting here, are part of the construction of the generic json which is the intermediary representation. I think of it like IR code for a compiler before a binary is created. Our IR, the generic json, is a superset of all the specific json options.

The source and options are used to build the generic json. From there, we build the specific json for the endpoint spec. We first pull from the generic json and then add the specific non generic pieces from the flags. In the end, we have the input data source's data in the correct format for endpoint plus whatever other options the user specified for the endpoint. The specific json is where you will find the options you are referencing above: Max-tokens, min-tokens, exclude_input_in_output, etc... 
The input file is strictly the prompt text, which I called out in the option help: The input file containing the single prompt to use for benchmarking.

Reply:
I see. I think you need to print a warning as well, since help options are easy to miss. I think it's particularly easy to miss the unstated implication that everything but input_text will get deleted.

The ideal solution would be a flow that allows all fields to be copied into the JSON IMO. I think we **need** a ticket for this, with the ticket number commented in this function. An intermediate solution would be to at least catch if the user provides a non-input_text field, and throw a warning in all caps, etc. Since we're so close to code freeze, I think adding a warning in the help message (e.g. All fields except input_text will be thrown out!) and adding a general error that is printed are fine. But we should create a ticket, link it here, and mention it in the next meeting so that this limitation is understood by everyone or it will bite us later.

Reply:
The ideal flow is exactly what we have today.
The user cannot control anything around the generic json except thru options on the command line.
We build the generic json programmatically, then we build the specific json with any extra data and details necessary for the specific endpoint from the generic json.

I think we are talking past one another here. The user gives a dataset, parameters to craft the synthetic data or now an input file with only the prompt text. The prompt is then put into the json file by our code.

In the PR description, I show an example of a prompt file and the resulting llm_inputs.json that is generated for a specific endpoint. The generic json is never shown because it is our intermediate representation. There is no way for the user to manipulate it beyond the options that are provided with the cli.

The user prompt text can be pretty much anything, we cant really warn the user about that.

Reply:
Ah, I'm following. Thanks for the explanation (and the visual in the PR file). I thought this would be a user-supplied option for supplying their own inputs via a JSON, but I missed the sample text file. So this is just a way for them to supply one prompt (which is the entire file) to use for the input. Got it, thanks for taking the time to clarify the goal of this change.